DEVELOPMENT.md

```diff
@@ -1 +1,179 @@
-See [the contributing guide](CONTRIBUTING.md).
\ No newline at end of file
+# Development
+
+## Getting Started
+
+1. [Create a GitHub Account](https://github.com/join)
+2. [Setup GitHub access via SSH](https://help.github.com/articles/connecting-to-github-with-ssh/)
+3. [Create and Checkout fork of this repository](#checkout-your-fork)
+4. [Install requirements](#requirements)
+5. [Two ways of running Hub](#two-ways-of-running)
+    - On your local machine
+    - On Kubernetes cluster
+
+### Checkout your fork
+
+To check out this repository:
+
+1. Create your own [fork of this repository](https://help.github.com/articles/fork-a-repo/)
+2. Clone it to your machine:
+```
+mkdir -p ${GOPATH}/src/github.com/tektoncd
+cd ${GOPATH}/src/github.com/tektoncd
+
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/hub.git
+cd hub
+
+git remote add upstream git@github.com:tektoncd/hub.git
+git remote set-url --push upstream no_push
+```
+
+Adding the upstream remote sets you up nicely for regularly [syncing your fork.](https://help.github.com/articles/syncing-a-fork/)
+
+### Requirements
+
+You must install these tools:
+
+1. [`go`](https://golang.org/doc/install): The language hub apis are built in.
+2. [`git`](https://help.github.com/articles/set-up-git/): For source control
+
+You may need to install more tools depending on the way you want to run the hub.
+
+### Two ways of Running
+
+- On local machine
+
+    For running on local machine you need postgresql database.
+
+    Two ways to run postgresql database:
+    - [Install postgresql](https://www.postgresql.org/docs/12/tutorial-install.html) on your local machine , or
+    - Run a postgresql container using [docker](https://docs.docker.com/engine/install/) / [podman](https://podman.io/getting-started/installation.html).
+
+    Run a postgresql container using below command:
+    ```
+    docker run -d  \
+    --name hub \
+    -e POSTGRES_USER=postgres \
+    -e POSTGRES_PASSWORD=postgres \
+    -e POSTGRES_DB=hub  \
+    -p 5432:5432 \
+    postgres
+    ```
+    Once you have postgresql instance running, you can run a migration to create tables and add data to tables.
+
+    **NOTE:** Make sure the database configurations are updated in [`hub/api/.env.dev`](https://github.com/tektoncd/hub/blob/master/api/.env.dev).
+
+    Ensure you are in `hub/api` directory.
+
+    Run the following command:
+    ```
+    go run ./cmd/db
+    ```
+    Wait for the logs to show
+    ```
+    DB initialisation successful !!
+    ```
+    Now, you can run api service:
+    ```
+    go run ./cmd/hub
+    ```
+    To run the **tests** you we need a test db instance:
+
+    If you have installed postgresql, just create a `hub_test` database. Otherwise, run a postgresql instance for `hub_test`.
+    ```
+    docker run -d  \
```

Comment:
need not run another container, you can use the same container; 
```
PGPASSWORD=$POSTGRES_PASSWORD psql -h localhost -p 5432 -U postgres -c 'create database hub;'
```
same for test db.